src/duckstation-qt/memoryscannerwindow.cpp

```diff
@@ -330,6 +332,22 @@ void MemoryScannerWindow::addManualWatchAddressClicked()
   updateWatch();
 }
 
+void MemoryScannerWindow::freezeWatchClicked()
+{
+  const int indexFirst = getSelectedWatchIndexFirst();
+  const int indexLast = getSelectedWatchIndexLast();
+  if (indexFirst < 0)
+    return;
+
+  const bool freeze = m_watch.GetEntryFreeze(indexFirst);
```

Comment:
I'm a little confused on this. It's effectively a toggle based on the first selected item.

Wouldn't you either way to implement it as a proper toggle, i.e. flip each entry depending on its state, or, just set them all to enabled?

Reply:
> I'm a little confused on this. It's effectively a toggle based on the first selected item.
> 
> Wouldn't you either way to implement it as a proper toggle, i.e. flip each entry depending on its state, or, just set them all to enabled?

Yes, its a toggle based on the first item since the real use case flow is usually either to mass enable or mass disable the freeze.
In real use you'd pretty much never want to independently flip with a mass selection, so rather than have dedicated 'freeze' and 'unfreeze' buttons, I use the first selected's state.
At the worst case you'd just have to press the button 3 times to sync your selection.

Reply:
Okay, if that makes sense for you, then sure.

I'm not using the feature, so don't have a particularly strong opinion either way.